frontend/internal-packages/jobs/src/trigger/deepModelingWorkflowTask.ts

```diff
@@ -39,34 +46,51 @@ export const deepModelingWorkflowTask = task({
     // Create fresh repositories in job to avoid serialization issues
     // When repositories are passed from API Route to Job, class instances lose their methods
     // during JSON serialization/deserialization, causing "createMessage is not a function" errors
-    const supabaseClient = createClient()
-    const repositories = createSupabaseRepositories(supabaseClient)
+    const supabaseClientResult = createClient()
 
-    const schemaResult = await repositories.schema.getSchema(
-      payload.designSessionId,
-    )
-    if (schemaResult.error || !schemaResult.data) {
-      throw new Error(`Failed to fetch schema data: ${schemaResult.error}`)
+    if (supabaseClientResult.isErr()) {
+      return errAsync(supabaseClientResult.error)
     }
+    const repositories = createSupabaseRepositories(supabaseClientResult.value)
 
     const workflowLogger = createWorkflowLogger()
 
-    const deepModelingParams: DeepModelingParams = {
-      ...payload,
-      schemaData: schemaResult.data.schema,
-    }
+    return ResultAsync.fromPromise(
+      repositories.schema.getSchema(payload.designSessionId),
+      (err) => new Error(`Failed to fetch schema data: ${String(err)}`),
+    )
+      .andThen((schemaResult) => {
+        if (!schemaResult.data) {
+          return errAsync(
+            new Error(
+              `Schema data is missing: ${
+                schemaResult.error ?? 'Unknown error'
+              }`,
+            ),
+          )
+        }
```

Comment:
This is a bit verbose, so could you simplify the caller side by changing the return value of `repositories.schema.getSchema()` to ResultAsync?